app/controllers/orders_controller.rb

```diff
@@ -1,13 +1,24 @@
 class OrdersController < ApplicationController
+  before_action :authenticate_user!
+
   def add_to_order
-    if Order.last
-      order = Order.last
-    else
-      order = Order.create
+	  if session[:order_id] == nil || Order.find(session[:order_id]).finalized == true
```

Comment:
If !session[:order_id] will only check if it's true or false. We want to see that it doesn'nt exist. We have tried with ! before and did not work. We therefore settled with the == nil. Cus that is what we want it to be in case you have not visited the site before and not created an order.